### PR TITLE
Add in implementation of standard deviation on data

### PR DESCRIPTION
**Pull request by thomaskileyukaea**
As per SR.1.1.1 add ability to calculate standard deviation for each day and then present this in the graphs show to the user.

### CODE REVIEW

File: inflammation-analysis.py [hunk 1]
```diff
@@ -21,7 +21,8 @@ def main(args):
     for filename in InFiles:
         inflammation_data = models.load_csv(filename)
 
-        view_data = {'average': models.daily_mean(inflammation_data), 'max': models.daily_max(inflammation_data), 'min': models.daily_min(inflammation_data)}
+        view_data = {'average': models.daily_mean(inflammation_data), 'max': models.daily_max(inflammation_data), 'min': models.daily_min(inflammation_data), **(models.s_dev(inflammation_data))}
```
Comment (thomaskileyukaea) [inline inflammation-analysis.py:24]: An example comment

File: inflammation/models.py [hunk 2]
```diff
@@ -31,3 +31,14 @@ def daily_max(data):
 def daily_min(data):
     """Calculate the daily min of a 2d inflammation data array."""
     return np.min(data, axis=0)
+
+
+def s_dev(data):
+    """Computes and returns standard deviation for data."""
+    number = np.mean(data, axis=0)
```
Comment (bielsnohr) [inline inflammation/models.py:38]: I don't immediately know what the purpose of the `number` variable is from its name. Based on its assignment, it is the mean along one axis, so perhaps it should be called `data_mean` or event just `mean`?

File: inflammation/models.py [hunk 3]
```diff
@@ -31,3 +31,14 @@ def daily_max(data):
...
+    devs = []
+    for entry in data:
+        devs.append((entry - number) * (entry - number))
+
+    s_dev2 = sum(devs) / len(data)
```
Comment (bielsnohr) [inline inflammation/models.py:43]: This is an implementation of standard deviation from the ground up. We could probably replace with a call to `np.std`.

File: inflammation/models.py [hunk 4]
```diff
@@ -31,3 +31,14 @@ def daily_max(data):
...
+    return {'standard deviation': s_dev2}
```
Comment (bielsnohr) [inline inflammation/models.py:44]: Returning a dict doesn't match what is returned for all of the other functions in this module. We should probably switch to just returning the calculated value.

File: inflammation/models.py [hunk 5]
```diff
@@ -31,3 +31,14 @@ def daily_max(data):
 def daily_min(data):
     """Calculate the daily min of a 2d inflammation data array."""
     return np.min(data, axis=0)
+
+
+def s_dev(data):
+    """Computes and returns standard deviation for data."""
```
Comment (thomaskileyukaea) [inline inflammation/models.py:37]: Doesn't return standard deviation directly, returns a dictionary

File: inflammation/models.py (hunk 4 above)
Comment (bielsnohr) [reply inflammation/models.py:44]: And it mixes in responsibilites that should be with the view

**Review by bielsnohr — CHANGES_REQUESTED** [review]
Overall, looking good. A few suggested changes to consider. Also, there are some tests that I would expect to see for this new feature:

- We should use some 2D data rather than just the 1D used here
- We should have some data with invalid values and check that the functions deal with these edge cases
- A test where the rows of a 2D matrix have been swapped around to check that the standard deviation is invariant (row swaps shouldn't affect its value).

File: inflammation/models.py (hunk 2 above)
Comment (thomaskileyukaea) [reply inflammation/models.py:38]: Resolved by using `np.std`
